### ISSUE TITLE
Test trigger acquisition logic

### ISSUE THREAD

**Issue by lukeryannetnz** [lukeryannetnz/quartznet-dynamodb#14]
The core logic around which triggers to fire, including the cluster locking/scheduler logic is contained in two large, complex methods:
`AcquireNextTriggers`
 `ApplyMisfireIfNecessary`

We need decent test coverage over these to assert they are behaving correctly before we can refactor/change them. It would also be nice to have confidence that the [dynamo conditional writes](http://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Expressions.SpecifyingConditions.html) used to [implement distributed locks](https://github.com/lukeryannetnz/quartznet-dynamodb/blob/master/src/QuartzNET-DynamoDB/JobStore.cs#L990-L1004) are working as desired.

**Comment by lukeryannetnz** [comment]
@ddhi004 if you want a challenge, this requires a fair amount of mental enterprise.
